src/Components/SysTable/SysTable.js

```diff
@@ -1,183 +1,173 @@
-import React, { useEffect, useReducer } from 'react';
+import React, { useContext, useRef } from 'react';
 import {
-    SortByDirection,
-    Table,
-    TableBody,
-    TableHeader,
     cellWidth,
     sortable
 } from '@patternfly/react-table/dist/esm/components/Table/index';
 
+import {
+    useFetchSystems,
+    useInventoryUtilities,
+    buildApiFilters
+} from './hooks';
+import { defaultOnLoad } from './constants';
+
 import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
+import { InventoryTable } from '@redhat-cloud-services/frontend-components/Inventory';
 import { GET_SYSTEM_TABLE } from '../../operations/queries';
 import { Link } from 'react-router-dom';
-import Loading from '../Loading/Loading';
 import MessageState from '../MessageState/MessageState';
-import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/PrimaryToolbar';
 import { SearchIcon } from '@patternfly/react-icons/dist/esm/icons/search-icon';
 import StatusLabel from '../StatusLabel/StatusLabel';
 import { Tooltip } from '@patternfly/react-core/dist/esm/components/Tooltip/Tooltip';
+import { Spinner } from '@patternfly/react-core';
 import { gqlProps } from '../Common';
 import messages from '../../Messages';
-import { sysTableFilters } from '../../store/cache';
 import { totalMatchesTitle } from '../Common';
 import { useIntl } from 'react-intl';
-import { useQuery } from '@apollo/client';
-import { useReactiveVar } from '@apollo/client';
-
-const initialState = {
-    tableVars: {
-        limit: 10,
-        offset: 0,
-        orderBy: 'LAST_SCAN_DATE_DESC',
-        name: ''
-    },
-    sortBy: {
-        index: 4,
-        direction: SortByDirection.desc
-    },
-    rows: []
-};
-const sortIndices = { 0: 'DISPLAY_NAME', 1: 'HAS_MATCH', 2: 'LAST_MATCH_DATE', 3: 'TOTAL_MATCHES', 4: 'LAST_SCAN_DATE' };
-const orderBy = ({ index, direction }) => `${sortIndices[index]}_${direction === SortByDirection.asc ? 'ASC' : 'DESC'}`;
-
-const tableReducer = (state, action) => {
-    switch (action.type) {
-        case 'setTableVars':
-            return { ...state, tableVars: { ...state.tableVars, ...action.payload } };
-        case 'setSortBy':
-            return { ...state, sortBy: action.payload, tableVars: { ...state.tableVars, ...action.tableVars } };
-        case 'setRows':
-            return { ...state, rows: action.payload };
-    }
-
-    return state;
-};
+import { RegistryContext } from '../../store';
 
 const SysTable = () => {
+    const inventory = useRef(null);
     const intl = useIntl();
-    const [{ tableVars, sortBy, rows }, stateSet] = useReducer(tableReducer, {
-        ...initialState
-    });
-    const { data: sysTableData, loading: sysTableLoading, error: sysTableError } =
-        useQuery(GET_SYSTEM_TABLE, { variables: { ...tableVars, ...useReactiveVar(sysTableFilters) } });
+    const { getRegistry } = useContext(RegistryContext);
+
     const columns = [
```

Comment:
Might be worth investigating moving this constant out of the component and see if everything still works. I couldn't find any state or prop variable so it might be worth moving it out of the component.